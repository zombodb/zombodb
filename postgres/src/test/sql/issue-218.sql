-- SQL
SELECT id FROM products WHERE (id=1 OR id=2 OR id=3) ORDER BY id;		-- 1, 2, 3 (correct)
-- zdb
SELECT id FROM products WHERE zdb('products', ctid) ==> '(id=1 OR id=2 OR id=3)' ORDER BY id;	-- 1, 2, 3 (correct)
SELECT id FROM products WHERE zdb('products', ctid) ==> '(id=1,id=2,id=3)' ORDER BY id;		-- 1, 2, 3 (correct)


-- SQL
SELECT id FROM products WHERE (id=1 AND id=2 AND id=3) ORDER BY id;		        -- empty result set (correct)
-- zdb
SELECT id FROM products WHERE zdb('products', ctid) ==> '(id=1 AND id=2 AND id=3)' ORDER BY id;  	-- 1, 2, 3 (wrong)
SELECT id FROM products WHERE zdb('products', ctid) ==> '(id=1 & id=2 & id=3)' ORDER BY id;		-- 1, 2, 3 (wrong)