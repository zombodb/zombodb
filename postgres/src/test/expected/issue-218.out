-- SQL
SELECT id FROM products WHERE (id=1 OR id=2 OR id=3) ORDER BY id;		-- 1, 2, 3 (correct)
 id 
----
  1
  2
  3
(3 rows)

-- zdb
SELECT id FROM products WHERE zdb('products', ctid) ==> '(id=1 OR id=2 OR id=3)' ORDER BY id;	-- 1, 2, 3 (correct)
 id 
----
  1
  2
  3
(3 rows)

SELECT id FROM products WHERE zdb('products', ctid) ==> '(id=1,id=2,id=3)' ORDER BY id;		-- 1, 2, 3 (correct)
 id 
----
  1
  2
  3
(3 rows)

-- SQL
SELECT id FROM products WHERE (id=1 AND id=2 AND id=3) ORDER BY id;		        -- empty result set (correct)
 id 
----
(0 rows)

-- zdb
SELECT id FROM products WHERE zdb('products', ctid) ==> '(id=1 AND id=2 AND id=3)' ORDER BY id;  	-- 1, 2, 3 (wrong)
 id 
----
(0 rows)

SELECT id FROM products WHERE zdb('products', ctid) ==> '(id=1 & id=2 & id=3)' ORDER BY id;		-- 1, 2, 3 (wrong)
 id 
----
(0 rows)

